sdks/python/apache_beam/runners/portability/portable_runner.py

```diff
@@ -217,11 +217,22 @@ def send_prepare_request(max_retries=5):
           staging_location='')
     else:
       retrieval_token = None
+
+    # Run the job and wait for a result.
     run_response = job_service.Run(
         beam_job_api_pb2.RunJobRequest(
             preparation_id=prepare_response.preparation_id,
             retrieval_token=retrieval_token))
-    return PipelineResult(job_service, run_response.job_id, cleanup_callbacks)
+
+    state_stream = job_service.GetStateStream(
```

Comment:
So, it seems we still have a bit of a race here (though it's as tight as can be). 

What if you tried getting the streams basted on the prepare_response.preparation_id, and only if that fails you fall back to getting them after starting the job? In the meantime, we can switch all runners to not have two separate ids (is there a JIRA for this? If not, please file one) and then remove the workaround.

Reply:
Sounds good, I made https://issues.apache.org/jira/browse/BEAM-6442 to follow-up on this. Will resolve when done

Reply:
Done

Reply:
Did you push?